controllers/constraintstatus_controller.go

```diff
@@ -0,0 +1,274 @@
+			},
+			Spec: v1alpha1.ConfigSpec{
+				Validation: v1alpha1.Validation{},
+				Sync: v1alpha1.Sync{
+					SyncOnly: []v1alpha1.SyncOnlyEntry{},
+				},
+			},
+		}
+		err = r.Create(ctx, config)
+
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	contraintStatusList := &v1beta1.ConstraintPodStatusList{}
+	err = r.List(ctx, contraintStatusList, client.InNamespace(request.NamespacedName.Namespace))
+
+	if err != nil {
+		return reconcile.Result{Requeue: false}, err
+	}
+
+	constraintKindNames := map[string][]string{}
+
+	for _, c := range contraintStatusList.Items {
+		if !slices.Contains(c.Status.Operations, "audit") {
+			continue
+		}
+
+		labels := c.GetLabels()
+
+		constraintKindNames[labels["internal.gatekeeper.sh/constraint-kind"]] = append(constraintKindNames[labels["internal.gatekeeper.sh/constraint-kind"]], ////nolint:lll
+			labels["internal.gatekeeper.sh/constraint-name"])
+	}
+
+	syncOnlys := []v1alpha1.SyncOnlyEntry{}
+
+	for kind, names := range constraintKindNames {
+		constraintGVR := schema.GroupVersionResource{
+			Group:    "constraints.gatekeeper.sh",
+			Version:  "v1beta1",
+			Resource: strings.ToLower(kind),
+		}
+		for _, name := range names {
+			constraint, err := r.DynamicClient.Resource(constraintGVR).Get(ctx, name, metav1.GetOptions{})
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			constraintKinds, _, err := unstructured.NestedSlice(constraint.Object, "spec", "match", "kinds")
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			r.addSyncOnlys(constraintKinds, &syncOnlys)
+		}
+	}
+
+	deleteDuplicateSyncOnly(&syncOnlys)
+	config.Spec.Sync.SyncOnly = syncOnlys
+	err = r.Update(ctx, config, &client.UpdateOptions{})
+
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func deleteDuplicateSyncOnly(syncOnlys *[]v1alpha1.SyncOnlyEntry) {
+	unique := []v1alpha1.SyncOnlyEntry{}
+
+	for _, sync := range *syncOnlys {
+		isUniq := true
+
+		for _, uniq := range unique {
+			if uniq.Group == sync.Group &&
+				uniq.Version == sync.Version &&
+				uniq.Kind == sync.Kind {
+				isUniq = false
+
+				break
+			}
+		}
+
+		if isUniq {
+			unique = append(unique, sync)
+		}
+	}
+
+	*syncOnlys = unique
+}
+
+func (r *ConstraintStatusReconciler) addSyncOnlys(constraintKinds []interface{}, syncOnlys *[]v1alpha1.SyncOnlyEntry) {
+	for _, kind := range constraintKinds {
+		newKind := kind.(map[string]interface{})
+		apiGroups := newKind["apiGroups"].([]interface{})
+		kindsInKinds := newKind["kinds"].([]interface{})
+
+		for _, apiGroup := range apiGroups {
+			for _, kindkind := range kindsInKinds {
+				version := r.getApiVersion(kindkind.(string), apiGroup.(string))
+
+				*syncOnlys = append(*syncOnlys, v1alpha1.SyncOnlyEntry{
+					Group:   apiGroup.(string),
+					Version: version,
+					Kind:    kindkind.(string),
+				})
+			}
+		}
+	}
+}
+
+func (r *ConstraintStatusReconciler) getApiVersion(kind string, apiGroup string) string {
```

Comment:
This is the preferred naming scheme in Go:
```suggestion
func (r *ConstraintStatusReconciler) getAPIVersion(kind string, apiGroup string) string {
```